src/v/raft/follower_stats.cc

```diff
@@ -32,10 +44,36 @@ void follower_stats::update_with_configuration(const group_configuration& cfg) {
           rni);
         it->second.is_learner = false;
     });
+    std::vector<container_t::node_type> removed;
 
-    absl::erase_if(_followers, [&cfg](const container_t::value_type& p) {
-        return !cfg.contains(p.first);
-    });
+    // removed
+    auto it = _followers.begin();
+    while (it != _followers.end()) {
+        if (cfg.contains(it->first)) {
+            ++it;
+        } else {
+            removed.push_back(_followers.extract(it++));
+        }
+    }
+
+    for (auto& n : removed) {
+        co_await n.mapped().requests_queue.stop();
+    }
```

Comment:
I think we should try to simplify this even further. Fundamentally we want a queue on sending requests, but handling the configuration updates means that a lot of code is changes in a very sensitive part of the code base that is slated for a refactoring in the near future. Just the futurization of a method could have consequences if we messed up some scheduling.

If I understand correctly why the configuration updates are pushed to the follower queue it is so that the semaphores are GCd when changes happen? If so:

One simplification would be to background `stop` while holding the consensus gate. That would mean all the call sides wouldn't need to be futurized. That would be a +1. But I think we can do even better:

If you have the following interface on the follower queue

```c++
    ss::future<std::optional<ss::semaphore_units<>>> get_append_entries_unit();
    void return_entries(ss::semaphore_units<>, vnode)
```

Then I think you could have something like this for `return_entries`:

```c++
1. release units
2. lookup sempahore
3. if semaphore has no waiters, delete semaphore
```

Then I don't think you would need to inform the follower queue of configuration changes. The throttling semaphores would simply be deleted when they are unused.

An even simpler solution would be to GC semaphores that have no waiters at a convenient time like in an existing timer, or even in send_append_entries (a short loop would look at a few integers max and free anything that's idle), or really any reliably periodic action.